### ISSUE TITLE
New buzzwords This Week in Programming: Before the AI Apocalypse

### ISSUE THREAD

**Issue by sudocurse** [sudocurse/reading-list#7879]
This Week in Programming: Before the AI Apocalypse<br>
http://ift.tt/2DgmWTt<br>
<img width="640" height="359" src="https://storage.googleapis.com/cdn.thenewstack.io/media/2018/03/8fd13a20-ubisoft.png" class="webfeedsFeaturedVisual wp-post-image wp-stateless-item" style="display: block; margin: auto; margin-bottom: 20px;" data-image-size="large" data-stateless-media-bucket="cdn.thenewstack.io" data-stateless-media-name="media/2018/03/8fd13a20-ubisoft.png">
<p>Before we get to that whole part where <a href="https://www.thenewstack.io/tag/artificial-intelligence" class="local-link">artificial intelligence</a> leads to machines coming alive, deciding humankind is an inferior virus that needs to be wiped from the face of the earth, and launches an unstoppable attack, well, it&rsquo;s worth keeping in mind that AI might just help us out a little bit too. &ldquo;Us&rdquo; being developers, at least.</p>
<p>This week, <a href="http://www.wired.co.uk/article/ubisoft-commit-assist-ai" class="ext-link" rel="external nofollow">a story in Wired</a> highlights how Ubisoft, a French gaming company, is using an AI assistant called &ldquo;Commit Assistant&rdquo; to help developers catch bugs before they ever make it into production.</p>
<p>This is the sort of AI I really expect to see a lot of, long before we ever make it to that whole apocalyptic doomsday thing. Remember that guy who <a href="https://workplace.stackexchange.com/questions/93696/is-it-unethical-for-me-to-not-tell-my-employer-i-ve-automated-my-job" class="ext-link" rel="external nofollow">automated his job</a>? Like that, but more complicated. Developers building tools to help themselves do what they do, but easier and better. That&rsquo;s what technology is all about, isn&rsquo;t it? And if you&rsquo;re going to build something, why not make it something that makes your own life easier?</p>
<p>According to Wired, they fed the AI ten years of code, &ldquo;allowing it to learn where mistakes have historically been made, reference any corrections that were applied, and predict when a coder may be about to write a similar bug.&rdquo; The main downside to the whole thing, so far, is that it apparently takes a whole lot of data and computing power to reach the point where it can identify six out of ten bugs, according to <a href="https://futurism.com/ai-predicts-coding-mistakes-before-developers-make-them/" class="ext-link" rel="external nofollow">another report</a> on the topic. But hey, computing power and data storage is getting cheaper by the day, right? Either way, instead of fearing that <a href="https://www.technologyreview.com/s/603381/ai-software-learns-to-make-ai-software/" class="ext-link" rel="external nofollow">AI is here to take your coding job</a>, perhaps just be happy that &mdash; until that whole life eradication thing&nbsp;&mdash; it&rsquo;s more likely here to help more than hurt&hellip; again, at least in terms of coding. Privacy, basic human dignity&nbsp;&mdash; all that stuff&nbsp;&mdash; well, that might be a whole other story.</p>
<p></p>
<h2>This Week in Programming</h2>
<ul>
<li>
<strong>Google Debuts Android P Developer Preview:</strong> Of course, with a dominant share of the smartphone OS market, Google&rsquo;s <a href="http://android-developers.googleblog.com/2018/03/previewing-android-p.html" class="ext-link" rel="external nofollow">debut of Android P</a> is the leading news this week for many developers out there. There are quite a few new features in the release, so we&rsquo;ll just outline a few. First, Android P will use WiFi RTT to provide accurate indoor positioning with the <a href="https://developer.android.com/reference/android/net/wifi/rtt/package-summary.html" class="ext-link" rel="external nofollow">RTT API</a>, giving you the ability to determine a user&rsquo;s indoor position with an accuracy of 1 to 2 meters. Next, Android P comes with a multi-camera API to access streams simultaneously from <a href="https://developer.android.com/reference/android/hardware/camera2/CameraMetadata.html#REQUEST_AVAILABLE_CAPABILITIES_LOGICAL_MULTI_CAMERA" class="ext-link" rel="external nofollow">two or more physical cameras</a>, ImageDecoder for bitmaps and drawables, several media APIs, neural networks API 1.1, improvements to autofill, and new mobile payments APIs. Beyond all of this, Android P also furthers the company&rsquo;s commitment to Kotlin, with compiler optimizations and continued work with Kotlin-creator JetBrains to optimize Kotlin&rsquo;s generated code. There&rsquo;s simply too much to go through here, so make sure to head over to <a href="http://android-developers.googleblog.com/2018/03/previewing-android-p.html" class="ext-link" rel="external nofollow">Google&rsquo;s blog post</a> to get the full details &mdash; or just watch the video and move on with your day.</li>
</ul>
<p></p>
<ul>
<li>
<strong>Android Things Developer Preview:</strong> Before we get carried away and move on from Google news, the latest <a href="http://android-developers.googleblog.com/2018/03/android-things-developer-preview-7.html" class="ext-link" rel="external nofollow">preview of Google&rsquo;s IOT platform</a> for developers, <a href="https://developer.android.com/things/preview/releases.html" class="ext-link" rel="external nofollow">Android Things Developer Preview 7</a> (DP7), was released this week as well and brings with it a slew of new features. DP7 is based on Android 8.1 and supports video and audio processing, as well as onboard machine learning with TensorFlow. In addition, it now supports increased camera resolution and MIDI and allows various new controls and analytics to support different hardware and software models, including separate update channels. DP7 also smooths out some inconsistent naming in the API and provides <a href="https://developer.android.com/things/sdk/apis/bluetooth.html" class="ext-link" rel="external nofollow">new Bluetooth APIs</a>.</li>
<li>
<strong>Here Comes Java 10:</strong> Not to be confused with EE4J or &ldquo;Jakarta,&rdquo; Java 10 is still under the parentage of Oracle and SDTimes has <a href="https://sdtimes.com/java/inside-look-java-10/" class="ext-link" rel="external nofollow">an inside look into Java 10</a>, as the latest version will be released this month &ldquo;with a focus on cloud and serverless computing.&rdquo; Now on a twice-yearly release schedule, the March release will include performance improvements (what release doesn&rsquo;t?), and features to make it better suited for serverless and cloud deployments, such as its new ability to share common class metadata across different Java processes. SDTimes also notes that Java 10 will continue its march toward feeling &ldquo;like a traditional functional language,&rdquo; while at the same time maintaining &ldquo;Java&rsquo;s commitment to static type safety and improving the developer experience by reducing the ceremony associated with writing Java code.&rdquo;</li>
<li>
<strong>GitHub Open Sources &ldquo;Licensed&rdquo; Tool:</strong> Dealing with code compliance and keeping things running in an open source project can be a major pain and GitHub has a solution for that &mdash; its <a href="https://github.com/github/licensed" class="ext-link" rel="external nofollow">Licensed</a> tool, which it just made <a href="https://githubengineering.com/improving-your-oss-dependency-workflow-with-licensed/" class="ext-link" rel="external nofollow">available as an open source project</a> itself this week. The tool &ldquo;helps GitHub engineers make efficient use of OSS by surfacing potential problems with a dependency&rsquo;s license early in our development cycle, ensuring we maintain dependency license documentation throughout our development cycle.&rdquo; And now it can help you do the same. As the company notes in its blog post, managing dependencies can be as easy as using a package manager that provides an easily parseable file of dependencies or as difficult as getting dirty with a command line tool. With Licensed, the tool does the dirty work by finding, caching and checking license metadata for dependencies across multiple language types and multiple projects in a single repository. For more details, give the Licensed <a href="https://github.com/github/licensed/blob/master/docs/configuration.md" class="ext-link" rel="external nofollow">documentation</a> a look-see.</li>
<li>
<strong>Visual Studio Updates:</strong> For you Visual Studio users out there, there are some new versions available &mdash; <a href="https://blogs.msdn.microsoft.com/visualstudio/2018/03/06/visual-studio-2017-version-15-6-visual-studio-for-mac-version-7-4-released/" class="ext-link" rel="external nofollow">Visual Studio 2017 version 15.6 and Visual Studio for Mac version 7.4</a>. The biggest thing to look for with this release is a performance improvement, with .NET Core users experiencing &ldquo;an average of 20 percent faster load times, with a more noticeable improvement for solutions with 30+ projects.&rdquo; Debugger&rsquo;s Threads also got a speed boost and now process in the background, and if your extensions are bogging things down, this version will help by adding notifications for extensions that may be causing UI delays. Go ahead and download <a href="http://aka.ms/vs-download?utm_source=vsblog&amp;utm_medium=post&amp;utm_campaign=15.6rtw" class="ext-link" rel="external nofollow">Visual Studio 2017 version 15.6</a> or <a href="https://aka.ms/vs4mac-download?utm_source=vsblog&amp;utm_medium=post&amp;utm_campaign=15.6rtw" class="ext-link" rel="external nofollow">Visual Studio for Mac</a> to get the latest.</li>
</ul>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">My career in a nutshell. <a href="https://t.co/02Iek65WqK" class="ext-link" rel="external nofollow">pic.twitter.com/02Iek65WqK</a></p>
<p>&mdash; Kelly Vaughn in NZ <img src="https://s.w.org/images/core/emoji/2.4/72x72/1f1f3-1f1ff.png" alt="&#127475;&#127487;" class="wp-smiley" style="height: 1em;"> (@kvlly) <a href="https://twitter.com/kvlly/status/971446313773486080?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">March 7, 2018</a></p>
</blockquote>
</div>
<ul>
<li>
<strong>GDPR for Developers:</strong> There are a lot of articles out there about the upcoming implementation of the <a href="https://www.eugdpr.org/" class="ext-link" rel="external nofollow">General Data Protection Regulation</a> (GDPR), but the majority of them are geared toward upper management types, telling them to hire people to handle the transition. And those people might be you. So, my ears perked up a little bit when I saw this <a href="https://techblog.bozho.net/gdpr-practical-guide-developers/" class="ext-link" rel="external nofollow">practical guide for developers</a> on how to handle GDPR requirements. The guide looks at aspects of GDPR that actually affect a developer and explains how they can address them &mdash; point by point. For example, there are several rights of the user that are relevant for developers, such as&rdquo;the right to erasure (the right to be forgotten/deleted from the system), right to restriction of processing (you still keep the data, but mark it as &lsquo;restricted&rsquo; and don&rsquo;t touch it without further consent by the user), the right to data portability (the ability to export one&rsquo;s data in a machine-readable format), the right to rectification (the ability to get personal data fixed), the right to be informed (getting human-readable information, rather than long terms and conditions), the right of access (the user should be able to see all the data you have about them).&rdquo; This guide goes through each one and explains how you, as the developer responsible for making them happen, can do so.</li>
</ul>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">Someone did this. I am in awe. <a href="https://t.co/UjPsElezAW" class="ext-link" rel="external nofollow">pic.twitter.com/UjPsElezAW</a></p>
<p>&mdash; Ian Yorston (@IanYorston) <a href="https://twitter.com/IanYorston/status/970289160643858432?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">March 4, 2018</a></p>
</blockquote>
</div>
<p class="attribution"><a href="https://thenewstack.io/sponsors/" class="local-link">Google</a> is a sponsor of The New Stack.</p>
<p>The post <a rel="nofollow" href="https://thenewstack.io/week-programming-ai-apocalypse/">This Week in Programming: Before the AI Apocalypse</a> appeared first on <a rel="nofollow" href="https://thenewstack.io">The New Stack</a>.</p>
<br>
<br>
via Blog &ndash; The New Stack https://thenewstack.io<br>
March 10, 2018 at 09:00AM
